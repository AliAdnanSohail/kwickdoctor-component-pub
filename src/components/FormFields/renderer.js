import React from 'react';
import classnames from 'classnames';

import styles, { error as errorStyles, label as labelStyles } from './styles';

export default render => ({
  input, meta, label, ...rest
}) => (
  <div className={classnames('form-field', rest.containerClassName)}>
    <label htmlFor={rest.id}>{label}</label>
    {render(input, label, {
      ...rest,
      className: classnames({
        invalid: meta && meta.error && meta.touched,
        active: meta && meta.active,
      }),
    })}
    {meta && meta.error && meta.touched ? <div className="error">{meta.error}</div> : undefined}

    <style jsx>{styles}</style>
    <style jsx>{labelStyles}</style>
    <style jsx>{errorStyles}</style>
  </div>
);
