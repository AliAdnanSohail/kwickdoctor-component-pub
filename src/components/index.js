export { default as AvatarPicker } from './AvatarPicker';
export { default as Button } from './Button';
export { default as ButtonGroup } from './ButtonGroup';
export { default as Calendar } from './Calendar';
export { default as Card } from './Card';
export { default as CardPicker } from './CardPicker';
export { default as DatePicker } from './DatePicker';
export { default as FileInput } from './FileInput';
export { default as Form } from './Form';
export { default as Modal } from './Modal';
export { default as Toast } from './Toast';
export { default as ToastManager } from './Toast/Manager';

export * from './FormFields';
export * from './Stepper';
