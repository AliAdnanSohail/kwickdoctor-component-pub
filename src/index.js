export { default as Button } from './components/Button';
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as Calendar } from './components/Calendar';
export { default as Card } from './components/Card';
export { default as Datepicker } from './components/Datepicker';
export { default as TextInput } from './components/TextInput';
export { default as Toast } from './components/Toast';
<<<<<<< HEAD
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as AvatarPicker } from './components/AvatarPicker';
=======
>>>>>>> master
