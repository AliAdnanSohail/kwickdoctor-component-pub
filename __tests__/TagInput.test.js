import React from 'react';
import renderer from 'react-test-renderer';

import { TagInput } from '../src';


const exampleTags = [
  {
    _id: '1',
    name: 'tag one',
  },
  {
    _id: '2',
    name: 'tag two',
  },
];

const suggestions = ['Pseudo', 'classes', 'are', 'bolted', 'on', 'to', 'selectors', 'specify', 'state', 'or'];

const baseProps = {
  id: 'tag-input-id',
  label: 'tag-input-label',
  tags: exampleTags,
  tagCreate(name) { return ({ name, _id: '' }); },
  suggestions,
  existingTags: suggestions,
};

it('TagInput renders correctly', () => {
  const element = renderer.create(<TagInput {...baseProps} />).toJSON();

  expect(element).toMatchSnapshot();
});
