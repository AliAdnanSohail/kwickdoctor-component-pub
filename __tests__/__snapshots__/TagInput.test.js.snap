// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagInput renders correctly 1`] = `
Array [
  <label
    className="jsx-4038665278"
    htmlFor="tag-input-id"
  >
    tag-input-label
  </label>,
  <div
    className="jsx-3032076080 tag-input"
    onClick={[Function]}
  >
    <ul
      className="jsx-3032076080 tag-input__tag-list"
    >
      <li
        className="jsx-4040469366 tag-input__add-existing-tag"
      >
        <button
          className="jsx-4040469366"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <svg
            aria-label="add"
            className="grommetux-control-icon grommetux-control-icon-add grommetux-control-icon--responsive"
            height="24px"
            role="img"
            version="1.1"
            viewBox="0 0 24 24"
            width="24px"
          >
            <path
              d="M12,22 L12,2 M2,12 L22,12"
              fill="none"
              stroke="#000"
              strokeWidth="2"
            />
          </svg>
          Add existing tag
        </button>
      </li>
      <li
        className="jsx-1684534872 tag-input__tag"
      >
        <span
          className="jsx-1684534872 tag-input__tag-wrapper"
        >
          <span
            className="jsx-1684534872 tag-input__tag-text"
          >
            tag one
          </span>
          <button
            className="jsx-1684534872"
            onClick={[Function]}
          >
            <svg
              aria-label="close"
              className="grommetux-control-icon grommetux-control-icon-close grommetux-control-icon--responsive"
              height="24px"
              role="img"
              version="1.1"
              viewBox="0 0 24 24"
              width="24px"
            >
              <path
                d="M3,3 L21,21 M3,21 L21,3"
                fill="none"
                stroke="#000"
                strokeWidth="2"
              />
            </svg>
          </button>
        </span>
      </li>
      <li
        className="jsx-1684534872 tag-input__tag"
      >
        <span
          className="jsx-1684534872 tag-input__tag-wrapper"
        >
          <span
            className="jsx-1684534872 tag-input__tag-text"
          >
            tag two
          </span>
          <button
            className="jsx-1684534872"
            onClick={[Function]}
          >
            <svg
              aria-label="close"
              className="grommetux-control-icon grommetux-control-icon-close grommetux-control-icon--responsive"
              height="24px"
              role="img"
              version="1.1"
              viewBox="0 0 24 24"
              width="24px"
            >
              <path
                d="M3,3 L21,21 M3,21 L21,3"
                fill="none"
                stroke="#000"
                strokeWidth="2"
              />
            </svg>
          </button>
        </span>
      </li>
    </ul>
    <div
      className="jsx-3032076080 tag-input__input"
    >
      <input
        className="jsx-3032076080"
        id="tag-input-id"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>,
]
`;
